src/Maestro/tests/Maestro.ScenarioTests/EndToEndFlowLogic.cs

```diff
@@ -0,0 +1,352 @@
+
+
+                        TestContext.WriteLine("Cloning target repo to prepare the target branch");
+                        TemporaryDirectory reposFolder = await CloneRepositoryAsync(targetRepoName);
+                        using (ChangeDirectory(reposFolder.Directory))
+                        {
+                            await using (await CheckoutBranchAsync(targetBranch))
+                            {
+                                TestContext.WriteLine("Adding dependencies to target repo");
+                                await AddDependenciesToLocalRepo(reposFolder.Directory, source1Assets.ToList(), targetRepoUri);
+                                await AddDependenciesToLocalRepo(reposFolder.Directory, source2Assets.ToList(), targetRepoUri);
+
+                                TestContext.WriteLine("Pushing branch to remote");
+                                await GitCommitAsync("Add dependencies");
+                                await using (await PushGitBranchAsync("origin", targetBranch))
+                                {
+                                    TestContext.WriteLine("Trigger the dependency update");
+                                    await TriggerSubscriptionAsync(subscription1Id.Value);
+                                    await TriggerSubscriptionAsync(subscription2Id.Value);
+
+                                    List<DependencyDetail> expectedDependencies = expectedDependenciesSource1.Concat(expectedDependenciesSource2).ToList();
+
+                                    TestContext.WriteLine($"Waiting on PR to be opened in {targetRepoUri}");
+
+                                    if (isAzDoTest)
+                                    {
+                                        await CheckBatchedAzDoPullRequest(source1RepoName, source2RepoName, targetRepoName, targetBranch, expectedDependencies, reposFolder.Directory);
+                                    }
+                                    else
+                                    {
+                                        await CheckBatchedGitHubPullRequest(targetBranch, source1RepoName, source2RepoName, targetRepoName, expectedDependencies, reposFolder.Directory);
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        public async Task NonBatchedFlowTestBase(string targetBranch, string channelName, bool isAzDoTest, bool allChecks = false, bool isCoherencyTest = false)
+        {
+            string targetRepoName = testRepo2Name;
+            string sourceRepoName = testRepo1Name;
+            string childRepoName = testRepo2Name;
+
+            if (isCoherencyTest)
+            {
+                targetRepoName = testRepo3Name;
+                sourceBranch = "coherency-tree";
+                sourceCommit = "cc1a27107a1f4c4bc5e2f796c5ef346f60abb404";
```

Comment:
Are these real hashes to something intersting? Or should we make them more obviously fake.

Reply:
Yes, they're specifically the head of the coherency branches. I'll add a comment.